packages/components/src/FilterBar/context/FilterBarContext.tsx

```diff
@@ -80,33 +77,38 @@ export const FilterBarProvider = <ValuesMap extends FiltersValues>({
         typeof newValue === "object" && JSON.stringify(newValue) === "{}"
           ? undefined
           : newValue
-      onValuesChange({
-        ...values,
-        [id]: validValue,
+      dispatch({
+        type: "update_values",
+        values: { ...values, [id]: validValue },
       })
     },
     showFilter: (id: keyof ValuesMap): void =>
       dispatch({ type: "activate_filter", id }),
     hideFilter: (id: keyof ValuesMap): void => {
       dispatch({ type: "deactivate_filter", id })
-      onValuesChange({ ...values, [id]: undefined })
     },
     getInactiveFilters: () => getInactiveFilters<ValuesMap>(state),
     clearAllFilters: () => {
       state.activeFilterIds.forEach(id => {
         if (mappedFilters[id].isRemovable)
           dispatch({ type: "deactivate_filter", id })
       })
-      onValuesChange({})
+      dispatch({ type: "update_values", values: {} })
     },
   } satisfies FilterBarContextValue<any, ValuesMap>
 
+  const shouldUpdateValues = filters.some(
+    ({ id }) => state.values[id] !== values[id]
+  )
```

Comment:
Maybe make this a bit more pure and pass it the state?

Reply:
You mean make a util and pass in `state` and `(consumer) values`?

Reply:
Nah, just like
```suggestion
  const shouldUpdateValues = (state) => filters.some(
    ({ id }) => state.values[id] !== values[id]
  )
```

Reply:
Lol, but I just want a boolean value :')

Reply:
Lol kay